dln/vi/utils.py

```diff
@@ -20,3 +24,118 @@ def compute_pairwise_kl(lps):
     return (
         (lps[:, :1, :] * (np.log(lps[:, :1, :]) - np.log(lps[:, :, :]))).sum(-1).mean(0)
     )
+
+class ResultLogEntry():
+    def __init__(self):
+        self.hiddens = None
+        self.candidates = [[], []]
+        self.metrics = {}
+        self.outputs = []
+
+    def log_metric(self, metric: str, value: float):
+        self.metrics[metric] = value
+
+    def log_outputs(self, outputs: str):
+        self.outputs = outputs
```

Comment:
Should `outputs:str` be appended to `self.outputs` or the `outputs` is `List[str]`?

Reply:
Thanks for catching the incorrect type hinting. The log_outputs stores the entire list or np.ndarray of strings returned by the VILModel's forward pass. So, the correct type would be List[str] or np.ndarray. But, since we might want to store and print any list-like structure in the form of outputs, I think it is better to remove the type hinting. (fixed in latest commit)